solrmonitor/solrmonitor.go

```diff
@@ -155,7 +155,8 @@ func (c *SolrMonitor) doGetCollectionState(name string) (*CollectionState, error
 	if coll == nil {
 		return nil, nil
 	}
-
+	coll.mu.RLock()
```

Comment:
This seems to add instead of remove a layer of locking. The above function locks.